### PR TITLE
Added tree list component

### PR DESCRIPTION
**Pull request by dlabrecq**
I'm able to build successfully and see template HTML in the webpack'ed file. Unit tests are included, but currently only run in the planner repo.

### CODE REVIEW

File: src/app/treelist/treelist.component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { Component, Input, TemplateRef, ViewChild, ViewEncapsulation } from '@angular/core';
+import { TreeNode, TREE_ACTIONS, KEYS, IActionMapping, TreeComponent } from 'angular2-tree-component';
+
+// See docs: https://angular2-tree.readme.io/docs
+@Component({
+  selector: 'alm-tree-list',
+  templateUrl: './treelist.component.html',
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: need to change this to template: require('./treelist.component.html'),

Same for styels. Take off "Url" and add require()

File: src/app/treelist/treelist.component.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  };
+
+  // Defaults
+  defaultOptions = {
+    actionMapping: this.actionMapping,
+    nodeClass: (node: TreeNode) => {
+      return 'tree-list';
+    }
+  }
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: missing semicolon

File: src/app/treelist/treelist.component.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      click: (tree, node, $event) => {
+        TREE_ACTIONS.TOGGLE_SELECTED(tree, node, $event)
+      }
+    }
+  };
+
+  // Defaults
+  defaultOptions = {
+    actionMapping: this.actionMapping,
+    nodeClass: (node: TreeNode) => {
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: the node param is not used

File: src/app/treelist/treelist.component.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      return 'tree-list';
+    }
+  }
+  customOptions = this.defaultOptions;
+
+  constructor() {
+    // Override default options
+    setTimeout(() => {
+      Object.assign(this.customOptions, this.options);
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: I don't think 'assign' is exists in Object. You might try 'call'

File: src/app/treelist/treelist.component.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { Component, Input, TemplateRef, ViewChild, ViewEncapsulation } from '@angular/core';
+import { TreeNode, TREE_ACTIONS, KEYS, IActionMapping, TreeComponent } from 'angular2-tree-component';
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: KEYS is not used

File: src/app/treelist/treelist.component.scss [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    padding: 0px;
+  }
+
+  // Toggle up
+  .toggle-children {
+    background-image: none;
+    background-position: inherit;
+    background-repeat: inherit;
+    display: block;
+    font-family: FontAwesome;
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.scss]: font-family doesn't have a generic default

File: src/app/treelist/treelist.component.ts (hunk 1 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.ts]: Will do

File: src/app/treelist/treelist.component.ts (hunk 3 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.ts]: Yes, it's used by the tree component

File: src/app/treelist/treelist.component.ts (hunk 4 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.ts]: I'm copying to a new object, so assign is correct

Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: my system complained about it. I'll take another look after you update the PR.

File: src/app/treelist/treelist.component.ts (hunk 5 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.ts]: Yes it is. The tree component uses this for keyboard functionality

Comment (joshuawilson) [inline src/app/treelist/treelist.component.ts]: ok

File: src/app/treelist/treelist.component.scss (hunk 6 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.scss]: This is the arrow for the toggle. What sort of default are you looking for?

Comment (dlabrecq) [inline src/app/treelist/treelist.component.scss]: It doesn't exist anywhere else

Comment (joshuawilson) [inline src/app/treelist/treelist.component.scss]: I was thinking that there should be a default incase FontAwesome wasn't there.

File: src/app/treelist/treelist.component.ts (hunk 4 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.ts]: I ran the build via the command line and in my IDE without errors. Is it the build or browser that's generating an error?

File: src/app/treelist/treelist.component.scss (hunk 6 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.scss]: The problem is that "\f105" won't match anything else. I checked Patternfly and they don't have a default either. 

.secondary-nav-item-pf {
  > a {
    cursor: default;
    &:after {
      content: @fa-var-angle-right;
      font-family: "FontAwesome";

File: src/app/treelist/treelist.component.ts (hunk 5 above)
Comment (dlabrecq) [inline src/app/treelist/treelist.component.ts]: Actually, KEYS can be removed now. I was using this for when the user hits enter, but removed the functionality.

keys: {
    [KEYS.ENTER]: (tree, node, $event) => alert(`This is ${node.data.name}`)
  }

**Pull request merged**